### PR TITLE
Ui cleanup

### PR DESCRIPTION
**Pull request by Neviothr**
**When merged this pull request will:**
- do some clean up in ui component.

### CODE REVIEW

File: addons/ui/flexiMenu/fnc_add.sqf [hunk 1]
```diff
@@ -11,9 +11,9 @@ _msg = "FLEXIMENU: Unknown Error in fnc_add.sqf"; //Initialize
 _key = [];
 
 // validate params
-_msg = format ["Error: invalid params. %1 (%2)", _this, __FILE__ ];  //
+_msg = format ["Error: invalid params. %1 (%2)", _this, __FILE__];
```
Comment (commy2) [inline addons/ui/flexiMenu/fnc_add.sqf]: Reminder for myself to double check this.

File: addons/ui/flexiMenu/fnc_getMenuOption.sqf [hunk 2]
```diff
@@ -5,20 +5,15 @@
...
+#define _flexiMenuSeparatorLine "<img image='\x\cba\addons\ui\flexiMenu\data\popup\separator.paa'/>" //<t size='1'> </t>  <t underline='true'>a    c</t>
 
-private["_menuDefs0", "_menuDef", "_fastPartialResult",
-    "_result", "_caption", "_action", "_actionOptions", "_icon", "_tooltip", "_subMenu", "_shortcut_DIK", "_visible", "_enabled",
-    "_array", "_index", "_containCaret", "_asciiKey", "_iconFolder", "_multiReselect",
-    "_keyName", "_offset", "_arrayID"];
-
-_menuDefs0 = _this select 0;
-_menuDef = _this select 1;
+private ["_menuDefs0", "_menuDef", "_fastPartialResult", "_result", "_caption", "_action", "_actionOptions", "_icon", "_tooltip", "_subMenu", "_shortcut_DIK", "_visible", "_enabled", "_array", "_index", "_containCaret", "_asciiKey", "_iconFolder", "_multiReselect", "_keyName", "_offset", "_arrayID"];
```
Comment (commy2) [inline addons/ui/flexiMenu/fnc_getMenuOption.sqf]: I liked this more broken up tbh.

File: addons/ui/flexiMenu/fnc_highlightCaretKey.sqf [hunk 3]
```diff
@@ -9,16 +9,10 @@ in order to avoid the millions of Deprecated function warnings:
...
 */
 
-private ['_array', '_index', '_offset', '_hotKeyColor_ST',
-    '_captionArray', '_i', '_len', '_array2'];
-
-_array = _this select 0;
-_index = _this select 1;
-_offset = _this select 2;
-_hotKeyColor_ST = _this select 3; // (ST=structured text) "<t color='#f07EB27E'>"
+private ["_captionArray", "_i", "_len", "_array2"];
```
Comment (commy2) [inline addons/ui/flexiMenu/fnc_highlightCaretKey.sqf]: The _i is set to private automatically by for FOR in A3.

File: addons/ui/flexiMenu/fnc_keyDown.sqf [hunk 4]
```diff
@@ -2,13 +2,10 @@
...
 
-#define _minObjDist(_var) (if (_var isKindOf "CAManBase") then {3} else {(2 max (1.4+(sizeOf typeOf _var)/2))}) // minimum object interaction distance: arbitrary distance. Might not work with very long/large vehicles. TODO: Find a very fast way to determine vehicle size.
+#define _minObjDist(_var) (if (_var isKindOf "CAManBase") then {3} else {(2 max (1.4 + (sizeOf typeOf _var) / 2))}) // minimum object interaction distance: arbitrary distance. Might not work with very long/large vehicles. TODO: Find a very fast way to determine vehicle size.
 
-private["_handled",
-    "_target", "_menuSource", "_active", "_potentialTarget", "_isTypeTarget",
-    "_potentialKeyMatch", "_potentialMenuSources", "_vehicleTarget", "_typesList",
-    "_keys", "_settings"];
-params["_ctrl","_dikCode","_shift","_ctrlKey","_alt"];
+private ["_handled", "_target", "_menuSource", "_active", "_potentialTarget", "_isTypeTarget", "_potentialKeyMatch", "_potentialMenuSources", "_vehicleTarget", "_typesList", "_keys", "_settings"];
```
Comment (commy2) [inline addons/ui/flexiMenu/fnc_keyDown.sqf]: Same as somewhere above.

File: addons/ui/flexiMenu/fnc_keyUp.sqf [hunk 5]
```diff
@@ -23,8 +18,7 @@ _potentialKeyMatch = false;
         if ((_x select 0 == _dikCode) &&
             {((!_shift && {!(_settings select 0)}) || {(_shift && {(_settings select 0)})})} &&
             {((!_ctrlKey && {!(_settings select 1)}) || {(_ctrlKey && {(_settings select 1)})})} &&
-            {((!_alt && {!(_settings select 2)}) || {(_alt && {(_settings select 2)})})} ) exitWith
-        {
+            {((!_alt && {!(_settings select 2)}) || {(_alt && {(_settings select 2)})})}) exitWith {
```
Comment (commy2) [inline addons/ui/flexiMenu/fnc_keyUp.sqf]: lol

File: addons/ui/flexiMenu/fnc_list.sqf [hunk 6]
```diff
@@ -3,16 +3,17 @@ disableSerialization;
 //-----------------------------------------------------------------------------
 #include "\x\cba\addons\ui\script_component.hpp"
 
-private ["_msg", "_menuSources", "_menuDefs", "_idc", "_iconFolder"];
-private["_menuOption", "_caption", "_action", "_icon", "_tooltip", "_shortcut_DIK", "_visible", "_enabled", "_array"];
+private ["_msg", "_menuSources", "_menuDefs", "_idc", "_iconFolder", "_menuOption", "_caption", "_action", "_icon", "_tooltip", "_shortcut_DIK", "_visible", "_enabled", "_array"];
```
Comment (commy2) [inline addons/ui/flexiMenu/fnc_list.sqf]: Same.

File: addons/ui/flexiMenu/fnc_list.sqf [hunk 7]
```diff
@@ -24,15 +25,16 @@ _iconFolder = if (count (_menuDefs select 0) > _flexiMenu_menuProperty_ID_iconFo
...
+if (isNil "_msg") then  {_msg = "FLEXIMENU: Unknown Error in fnc_list.sqf"};
 if (typeName _menuRsc != typeName "") exitWith {diag_log _msg};
 
 if (!isClass (configFile >> _menuRsc) && {!isClass (missionConfigFile >> _menuRsc)}) then { // if not a full class name
     _menuRsc = __menuRscPrefix + _menuRsc; // attach standard flexi menu prefix
 };
 
 // TODO: Support missionConfigFile too
-_width = getNumber(ConfigFile >> _menuRsc >> "flexiMenu_subMenuCaptionWidth");
+_width = getNumber (ConfigFile >> _menuRsc >> "flexiMenu_subMenuCaptionWidth");
```
Comment (commy2) [inline addons/ui/flexiMenu/fnc_list.sqf]: configFile

File: addons/ui/flexiMenu/fnc_menu.sqf [hunk 8]
```diff
@@ -82,16 +82,15 @@ Note: visible allows value -1 (instead of 0) to make the current button be re-us
 ]
 */
 //-----------------------------------------------------------------------------
-private ["_msg", "_valid", "_menuSources", "_menuDefs", "_menuParams", "_menuRsc", "_array", "_i",     "_t", "_w", "_idcIndex", "_idc"];
-private ["_caption", "_action", "_icon", "_subMenu", "_tooltip", "_shortcut_DIK", "_visible", "_enabled"];
-private ["_params", "_useListBox", "_menuOption", "_commitList", "_source", "_width", "_list"];
+private ["_msg", "_valid", "_menuSources", "_menuDefs", "_menuParams", "_menuRsc", "_array", "_i", "_t", "_w", "_idcIndex", "_idc", "_caption", "_action", "_icon", "_subMenu", "_tooltip", "_shortcut_DIK", "_visible", "_enabled", "_params", "_useListBox", "_menuOption", "_commitList", "_source", "_width", "_list"];
```
Comment (commy2) [inline addons/ui/flexiMenu/fnc_menu.sqf]: Same.

File: addons/ui/flexiMenu/fnc_menuShortcut.sqf [hunk 9]
```diff
@@ -5,15 +5,8 @@
...
-private['_handled', '_EHParams', '_control', '_dikCode', '_shiftKey', '_ctrlKey', '_altKey'];
-private['_menuDefs', "_caption", "_action", "_icon", "_tooltip", "_subMenu", "_shortcut", "_visible", "_enabled"];
-_EHParams = _this select 0;
-
-_control = _EHParams select 0;
-_dikCode = _EHParams select 1;
-_shiftKey = _EHParams select 2;
-_ctrlKey = _EHParams select 3;
-_altKey = _EHParams select 4;
+private ["_handled", "_menuDefs", "_caption", "_action", "_icon", "_tooltip", "_subMenu", "_shortcut", "_visible", "_enabled"];
```
Comment (commy2) [inline addons/ui/flexiMenu/fnc_menuShortcut.sqf]: Same.

File: addons/ui/flexiMenu/fnc_menuShortcut.sqf [hunk 10]
```diff
@@ -5,15 +5,8 @@
...
+param [0] params ["_control", "_dikCode", "_shiftKey", "_ctrlKey", "_altKey"];
```
Comment (commy2) [inline addons/ui/flexiMenu/fnc_menuShortcut.sqf]:
```sqf
(_this select 0) params ...
```

Comment (Neviothr) [reply addons/ui/flexiMenu/fnc_menuShortcut.sqf]: You taught me this :unamused:

File: addons/ui/flexiMenu/fnc_add.sqf (hunk 1 above)
Comment (Neviothr) [reply addons/ui/flexiMenu/fnc_add.sqf]: Well?

File: addons/ui/flexiMenu/fnc_menuShortcut.sqf (hunk 10 above)
Comment (commy2) [reply addons/ui/flexiMenu/fnc_menuShortcut.sqf]: Don't take everything I say seriously.

**Pull request merged**